### PR TITLE
[cmd/builder] Remove undocumented strictness check

### PR DESCRIPTION
**Pull request by mx-psi**
#### Description

Partially reverts #9897. This was not documented on the original PR and is IMO too strict. 
We likely want to allow for some skew between versions.

Mentioned on https://github.com/open-telemetry/opentelemetry-collector/pull/9513#issuecomment-2065586307

### CODE REVIEW

File: Makefile.Common [hunk 1]
```diff
@@ -4,7 +4,7 @@ ALL_PKGS := $(sort $(shell go list ./...))
 # COVER_PKGS is the list of packages to include in the coverage
 COVER_PKGS := $(shell go list ./... | tr "\n" ",")
 
-GOTEST_TIMEOUT?=120s
+GOTEST_TIMEOUT?=240s
```
Comment (evan-bradley) [inline Makefile.Common:7]: If we're seeing timeouts in all our tests, we can probably remove the manual increase in the Windows unit tests workflow, which I just bumped to 240s as well.

**Review by evan-bradley — APPROVED** [review]

File: Makefile.Common (hunk 1 above)
Comment (mx-psi) [reply Makefile.Common:7]: Yeah I was testing if the timeout would help (it did), so I made your change from #9992 apply to every test suite

**Review by kristinapathak — APPROVED** [review]

**Review by codeboten — COMMENTED** [review]
Thanks @mx-psi, please resolve the conflict

File: Makefile.Common (hunk 1 above)
Comment (codeboten) [reply Makefile.Common:7]: It's pretty small, but it would be better if this change was not in this PR :)

**Review by codeboten — APPROVED** [review]

**Pull request merged**